rosidl_generator_cpp/resource/msg__traits.hpp.em

```diff
@@ -67,6 +71,32 @@ template<>
 struct has_fixed_size<@(cpp_namespace)@(spec.base_type.type)>
   : std::integral_constant<bool, @(fixed_template_string)>{};
 
+@{
+bounded_template_strings = []
+bounded = True
```

Comment:
Instead of setting the flag within the loop this could be written like this:

```
bounded = False:
for field in spec.fields:
    if ...:
        break
    if ...:
        break
    if field.type.is_primitive_type():
        # append to bounded_template_strings
```

That might make it more obvious when the variable is `True` and `False` without having to check where it is assigned within the loop.

[The same could be applied to the existing loop.]